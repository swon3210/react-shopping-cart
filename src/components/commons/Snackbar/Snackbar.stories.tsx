import Snackbar, { Props } from './Snackbar';

export default {
  title: 'Components/Commons/Snackbar',
  component: Snackbar,
  argTypes: {},
};

const Template = (args: Props) => <Snackbar {...args}>test</Snackbar>;

export const Default = Template.bind({});

(Default as any).args = {
  duration: 700,
};
