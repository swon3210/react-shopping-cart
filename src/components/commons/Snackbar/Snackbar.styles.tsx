import styled, { keyframes, css } from 'styled-components';

const moveUp = keyframes`
  from {
    transform: translateY(100px);
  }
  to {
    transform: translateY(-10px);
  }
`;

const moveDown = keyframes`
  from {
    transform: translateY(-10px);
  }
  to {
    transform: translateY(100px);
  }
`;

const moveUpAnimation = (duration: number) => css`
  animation: ${moveUp} ${duration / 1000}s forwards;
`;

const moveDownAnimation = (duration: number) => css`
  animation: ${moveDown} ${duration / 1000}s forwards;
`;

export const Snackbar = styled.div<{ isShown: boolean; duration: number }>`
  display: inline-block;
  position: fixed;
  font-size: 20px;
  line-height: 24px;
  color: ${({ theme }) => theme.WHITE};
  bottom: 0;
  left: 50%;
  transform: translateX(-50%);
  min-width: 270px;
  padding: 25px 20px;
  border-radius: 5px;
  background-color: ${({ theme }) => theme.BLACK_500};
  ${({ isShown, duration }) => (isShown ? moveUpAnimation(duration) : moveDownAnimation(duration))};
`;
