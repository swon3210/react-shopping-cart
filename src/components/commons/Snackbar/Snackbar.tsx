import React from 'react';
import * as Styled from './Snackbar.styles';

export interface Props {
  children: React.ReactNode;
  isShown: boolean;
  /** 밀리 초 단위로 입력해주십시오 */
  duration: number;
}

const Snackbar = ({ children, isShown, duration }: Props) => {
  return (
    <Styled.Snackbar isShown={isShown} duration={duration}>
      {children}
    </Styled.Snackbar>
  );
};

export default Snackbar;
