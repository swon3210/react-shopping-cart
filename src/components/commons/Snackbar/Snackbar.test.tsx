import Snackbar from './Snackbar';
import { render } from '@testing-library/react';

describe('Snackbar Component', () => {
  it('Snackbar Snapshot', () => {
    const snackbarUtil = render(
      <Snackbar isShown={true} duration={700}>
        TEST
      </Snackbar>
    );
    expect(snackbarUtil.asFragment()).toMatchSnapshot();
  });
});
