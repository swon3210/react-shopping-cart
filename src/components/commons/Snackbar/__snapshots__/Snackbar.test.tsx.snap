// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snackbar Component Snackbar Snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-bdvvaa kUMFhO"
  >
    TEST
  </div>
</DocumentFragment>
`;
